tools/shared.py

```diff
@@ -2853,6 +2871,18 @@ def get_binaryen_bin():
     assert Settings.WASM, 'non wasm builds should not ask for binaryen'
     return os.path.join(BINARYEN_ROOT, 'bin')
 
+  @staticmethod
+  def get_binaryen_command(tool, args=[], debug=False):
+    cmd = [os.path.join(Building.get_binaryen_bin(), tool)] + args
+    cmd += Building.get_binaryen_feature_flags()
+    if debug:
+      cmd += ['-g'] # preserve the debug info
+    return cmd
+
+  @staticmethod
+  def get_wasm_opt_command(args=[], debug=False):
```

Comment:
can you have this take and infile and outfile argument since all the call sites want that I think the command requires it?

Reply:
Sounds goos, but perhaps as a followup? That's going to be a bunch more changes not directly relevant to this PR.

Reply:
How about splitting the `get_wasm_opt_command` part out of this change?  Seems like a refactor that would make this change smaller.  I leave that up to you though, I know you wan to get this landed today if possible so i don't want to delay you too much.

Reply:
Well, the reason I added it here is that it made this shorter to write, and also seemed like a useful change. I don't see a benefit to splitting it out, but I agree this can be improved more, I'll look at that as a followup.